bin/stringsCompletenessTest.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/python
+
+##
+# This script is designed to validate string usage code completeness in the l10n property files and for
+# the standalone loop client and loop add-on executable code.
+# The loop repo is assumed to be https://github.com/mozilla/loop.
+#
+# Run this script from the local version of loop. It assumes that a local
+# version is in directory: ./loop/bin.
+##
+
+from __future__ import print_function
+
+import argparse
+import io
+import os
+import itertools as it
+import glob
+import re
+
+# defaults
+DEF_PROPERTY_LOCALE = "en-US"
+
+
+def main(l10n_locale):
+    # capture root path assuming this file is in loop/bin
+    # uncomment the next line if you are not running this from Makefile strings call
+    # os.chdir(os.path.abspath(os.pardir))
+    loop_root_dir = os.getcwd()
+    def_property_dirs = ["add-on", "shared", "standalone"]
+    # loop/locale/en-US
+    def_us_locale_dir = os.path.abspath(os.path.join(loop_root_dir, "locale", l10n_locale))
+    def_property_addon_file_name = os.path.join(def_us_locale_dir, os.extsep.join([def_property_dirs[0], "properties"]))
+    def_property_shared_file_name = os.path.join(def_us_locale_dir, os.extsep.join([def_property_dirs[1], "properties"]))
+    def_property_standalone_name = os.path.join(def_us_locale_dir, os.extsep.join([def_property_dirs[2], "properties"]))
+
+    print("l10n_locale", l10n_locale)
+    print(def_us_locale_dir)
+    print(def_property_addon_file_name)
```

Comment:
Most of this debug needs either simplification or removal or not displaying by default. Its too difficult to look at the output at the moment.